legend-engine-core-executionPlan-execution/legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/parameterization/ParameterizedValueSpecification.java

```diff
@@ -0,0 +1,391 @@
+
+    {
+        Map<String, BiFunction<AppliedFunction, String, List<ValueSpecification>>> parameterHelper = new HashMap<>();
+        parameterHelper.put("in", (fn, parameterName) -> Arrays.asList(fn.parameters.get(0), ManyParam(parameterName, fn.parameters.get(1))));
+        parameterHelper.put("filter", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, true, fn.function));
+        parameterHelper.put("take", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, true, fn.function));
+        parameterHelper.put("getAll", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, true, fn.function));
+        parameterHelper.put("limit", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, true, fn.function));
+
+        parameterHelper.put("col", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("project", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("groupBy", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("serialize", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("graphFetch", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("sort", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("tdsContains", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("restrict", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("join", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("objectReferenceIn", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("restrictDistinct", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("projectWithColumnSubset", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("groupByWithWindowSubset", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
+        parameterHelper.put("olapGroupBy", (fn, parameterName) -> valueSpecificationListHelper(fn.parameters, parameterName, false, fn.function));
```

Comment:
Rather than hardcoding these, is there a way we can enhance [Handlers](https://github.com/finos/legend-engine/blob/master/legend-engine-core-language-pure/legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/handlers/Handlers.java)to do this or help us with the lookups?

Reply:
At the moment this is probably the best place given that handlers  is part of the compiler and we don't want to compile this.